### ISSUE TITLE
Impact of num_classes

### ISSUE THREAD

**Issue by ykwongaq** [isaaccorley/detcon-pytorch#5]
Hi, thank you for providing the pytorch code for DetConB.

I want to ask how the num_classes in the config file affect the model training. From my understanding, no matter what number you set for the num_classes, the mask pooling layer will eventually collect 16 samples based on whether it exist or not, such that num_classes should not affect the final result, unless it is too low (eg < 16). 

For example, after the pooling the mask, if there are only 5 mask id exist in the mask, then the sampling should sample out these mask ids, together with other 11 non-existing mask id. Therefore, the num_classes should not affect the result.

However, based on what I observerd. When you set num_classes = 21, the initial loss will ~9. And when you set num_classes = 256 (because FH algorithm may generated more than 21 mask id), the initial loss will drop to ~3. Therefore, I want to ask how the num_classes impact the loss or is there any misunderstanding?
